Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/rolling_updater_test.go

```diff
@@ -328,199 +303,275 @@ Scaling foo-v2 up to 8
 Scaling foo-v1 down to 2
 Scaling foo-v2 up to 10
 Scaling foo-v1 down to 0
-Update succeeded. Deleting foo-v1
 `,
 		}, {
-			oldRc:    oldRc(2),
-			newRc:    newRc(6, 6),
-			accepted: true,
-			percent:  Percent(50),
-			responses: []fakeResponse{
-				// no existing newRc
-				{nil, fmt.Errorf("not found")},
-				// scaling iteration
-				{newRc(3, 6), nil},
-				{oldRc(0), nil},
-				// scaling iteration
-				{newRc(6, 6), nil},
-				// cleanup annotations
-				{newRc(6, 6), nil},
-				{newRc(6, 6), nil},
-				{newRc(6, 6), nil},
+			name:          "2/6/50 simple readiness",
+			oldRc:         oldRc(2),
+			newRc:         newRc(0, 6),
+			newRcExists:   false,
+			updatePercent: Percent(50),
+			expected: []interface{}{
+				up{3},
+				down{oldReady: 2, newReady: 1, to: 0},
+				up{6},
 			},
-			output: `Creating foo-v2
-Scaling up foo-v2 from 0 to 6, scaling down foo-v1 from 2 to 0 (scale up first by 3 each interval)
+			output: `Created foo-v2
+Scaling up foo-v2 from 0 to 6, scaling down foo-v1 from 2 to 0 (scale up first by 3 each interval, maintain at least 1 ready)
 Scaling foo-v2 up to 3
 Scaling foo-v1 down to 0
 Scaling foo-v2 up to 6
-Update succeeded. Deleting foo-v1
 `,
-		}, {
-			oldRc:    oldRc(10),
-			newRc:    newRc(3, 3),
-			accepted: true,
-			percent:  Percent(50),
-			responses: []fakeResponse{
-				// no existing newRc
-				{nil, fmt.Errorf("not found")},
-				// scaling iteration
-				{newRc(2, 3), nil},
-				{oldRc(8), nil},
-				// scaling iteration
-				{newRc(3, 3), nil},
-				{oldRc(0), nil},
-				// cleanup annotations
-				{newRc(3, 3), nil},
-				{newRc(3, 3), nil},
-				{newRc(3, 3), nil},
+		}, {name: "10/3/50 simple readiness",
+			oldRc:         oldRc(10),
+			newRc:         newRc(0, 3),
+			newRcExists:   false,
+			updatePercent: Percent(50),
+			expected: []interface{}{
+				up{2},
```

Comment:
So to get this clear, the SLA is defined based on the difference set between the desired replicas of the first controller that will be scaled (newRc is we start up, oldRc if we start down) and (desired*update percentage), right?